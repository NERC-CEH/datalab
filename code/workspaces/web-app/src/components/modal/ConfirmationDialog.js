import React from 'react';
import PropTypes from 'prop-types';
import Dialog from '@mui/material/Dialog';
import DialogTitle from '@mui/material/DialogTitle';
import DialogContent from '@mui/material/DialogContent';
import DialogActions from '@mui/material/DialogActions';
import Typography from '@mui/material/Typography';
import SecondaryActionButton from '../common/buttons/SecondaryActionButton';
import PrimaryActionButton from '../common/buttons/PrimaryActionButton';

const ConfirmationDialog = ({ onSubmit, title, body, onCancel }) => (
  <Dialog open={true} maxWidth="md">
    <DialogTitle>{title}</DialogTitle>
    <DialogContent>
      <Typography variant="body1">{body}</Typography>
    </DialogContent>
    <DialogActions>
      <PrimaryActionButton onClick={onSubmit} icon="check">Confirm</PrimaryActionButton>
      <SecondaryActionButton onClick={onCancel} icon="clear">Cancel</SecondaryActionButton>
    </DialogActions>
  </Dialog>
);

ConfirmationDialog.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
};

export default ConfirmationDialog;
