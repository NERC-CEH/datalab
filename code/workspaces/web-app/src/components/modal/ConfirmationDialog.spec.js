import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationDialog from './ConfirmationDialog';

describe('ConfirmationDialog', () => {
  const shallowRender = props => shallow(
    <ConfirmationDialog {...props} />,
  );

  it('renders to match snapshot', () => {
    const render = shallowRender({
      onSubmit: jest.fn().mockName('onSubmit'),
      title: 'Dialog title',
      body: 'Dialog body',
      onCancel: jest.fn().mockName('onCancel'),
    });

    expect(render).toMatchSnapshot();
  });
});
