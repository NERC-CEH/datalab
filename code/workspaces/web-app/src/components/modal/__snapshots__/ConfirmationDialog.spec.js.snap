// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationDialog renders to match snapshot 1`] = `
<div
  aria-labelledby="mui-1"
  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd css-rx25b2-MuiPaper-root-MuiDialog-paper"
  role="dialog"
>
  <h2
    class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-xq3hxg-MuiTypography-root-MuiDialogTitle-root"
    id="mui-1"
  >
    Dialog title
  </h2>
  <div
    class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-12gb3o9-MuiTypography-root"
    >
      Dialog body
    </p>
  </div>
  <div
    class="MuiDialogActions-root MuiDialogActions-spacing css-hlj6pa-MuiDialogActions-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-o0h1hg-MuiButtonBase-root-MuiButton-root"
      icon="check"
      tabindex="0"
      type="button"
    >
      Confirm
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-4eo762-MuiButtonBase-root-MuiButton-root"
      icon="clear"
      tabindex="0"
      type="button"
    >
      Cancel
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
